OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -441,5 +449,281 @@ public void TestGetVariationWithBucketingId()
+            var expectedExperimentId = featureFlag.ExperimentIds[0];
+            var expectedExperiment = ProjectConfig.GetExperimentFromId(expectedExperimentId);
+            var expectedVariation = expectedExperiment.Variations[0];
+
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureExperiment(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns(expectedVariation);
+
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeature(featureFlag, "user1", new UserAttributes());
+
+            Assert.AreEqual(expectedVariation, actualVariation);
+        }
+
+        // Should return the bucketed variation and null experiment 
+        [Test]
+        public void TestGetVariationForFeatureWhenTheUserIsNotBucketedIntoFeatureExperimentAndBucketedToFeatureRollout()
+        {
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("string_single_variable_feature");
+            var rolloutId = featureFlag.RolloutId;
+            var rollout = ProjectConfig.GetRolloutFromId(rolloutId);
+            var expectedExperiment = rollout.Experiments[0];
+            var expectedVariation = expectedExperiment.Variations[0];
+
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureExperiment(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureRollout(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns(expectedVariation);
+
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeature(featureFlag, "user1", new UserAttributes());
+
+            Assert.AreEqual(expectedVariation, actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is bucketed into a rollout for feature flag \"string_single_variable_feature\"."));
+        }
+
+        [Test]
+        public void TestGetVariationForFeatureWhenTheUserIsNeitherBucketedIntoFeatureExperimentNorToFeatureRollout()
+        {
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("string_single_variable_feature");
+
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureExperiment(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureRollout(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeature(featureFlag, "user1", new UserAttributes());
+            Assert.IsNull(actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is not bucketed into a rollout for feature flag \"string_single_variable_feature\"."));
+        }
+
+        #endregion // GetVariationForFeature Tests
```

Comment:
add one more test to make sure the user is bucketed into the experiment's variation when the user satisfies bucketing and traffic allocation for feature flag experiment and feature flag rollout

Reply:
Test named TestGetVariationForFeatureWhenTheUserIsBuckedtedInBothExperimentAndRollout has been added.